mixpanel/mixpanel_export_test.go

```diff
@@ -0,0 +1,75 @@
+package mixpanel
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+	"testing"
+	"time"
+)
+
+type marshalTestPair struct {
+	value    interface{}
+	strvalue string
+}
+
+func TestStringOrJsonMarshal(t *testing.T) {
+	tests := []marshalTestPair{
+		{"teststring", "teststring"},
+		{map[string]string{"key1": "val1", "key2:": "val2"}, "{\"key1\":\"val1\",\"key2:\":\"val2\"}"},
+		{[]string{"arrary", "of", "strings"}, "[\"arrary\",\"of\",\"strings\"]"},
+	}
+	for _, pair := range tests {
+		v, err := StringOrJsonMarshal(pair.value)
+		if err != nil {
+			t.Errorf("Unexpected error %s for StringOrJsonMarshal(%+v). Expected %s", err.Error(), pair.value, pair.strvalue)
+		}
+		if v != pair.strvalue {
+			t.Errorf("Unexpected value %s for StringOrJsonMarshal(%+v). Expected %s", v, pair.value, pair.strvalue)
+		}
+	}
+}
+
+func TestComputeSig(t *testing.T) {
+	params := map[string]interface{}{
+		"api_key": "TESTMIXPANELAPIKEY",
+		"expire":  12312324324,
+		"events":  []string{"event1", "event2"},
+		"param1":  "value1",
+	}
+	m, exp := MixpanelExport{apiSecret: "TESTMIXPANELAPISECRET"}, "1beb8d4f61da24302844e252a8ff6e75"
+	sig, err := m.ComputeSig(params)
+	if err != nil {
+		t.Errorf("Unexpected error %s for ComputeSign(%+v). Expected %s", err.Error(), params, exp)
+	}
+	if sig != exp {
+		t.Errorf("Unexpected value %s for ComputeSign(%+v). Expected %s", sig, params, exp)
+	}
```

Comment:
Could be `assert.Equal`.

One reason I push for this library is readable output, including printing of the expected and actual values